documentation/puppet-api/v3/environment_classes.md

```diff
@@ -270,117 +304,107 @@ The environment must be purely alphanumeric, not 'bog|us'
+latest available data is returned to every request.
+
+#### Behaviors when the environment class cache is enabled
+
+When the `environment-class-cache-enabled` setting is set to `true`, the response to a
+query to the `environment_classes`  endpoint includes an HTTP [Etag][] header. The value
+for the Etag header is a hash which represents the state of the latest class information
+available for the requested environment. For example:
 
 ```
 ETag: 31d64b8038258202b4f5eb508d7dab79c46327bb
 ```
 
-A client may (but is not required to) provide the Etag value back to the server
-in a subsequent `environment_classes` request.  The client would provide the
-tag value as the value for an
-[If-None-Match](https://tools.ietf.org/html/rfc7232#section-3.2)
-HTTP header:
+A client can (but is not required to) provide the Etag value back to the server in a
+subsequent `environment_classes` request. The client would provide the tag value as the
+value for an [If-None-Match](https://tools.ietf.org/html/rfc7232#section-3.2) HTTP header:
 
 ```
 If-None-Match: 31d64b8038258202b4f5eb508d7dab79c46327bb
 ```
 
-If the latest state of code available on the server matches that of the value
-in the `If-None-Match` header, the server will provide an HTTP 304 (Not
-Modified) response and no corresponding response body.  If the server has later
-code available than what is captured by the `If-None-Match` header value or
-no `If-None-Match` header is provided in the request, the server will re-parse
-code from manifest files on disk.  Assuming the resulting payload were different
-than in a previous request, the server would provide a different Etag value
-along new class information in the response payload.
+If the latest state of code available on the Server matches that of the value in the
+`If-None-Match` header, the Server returns an HTTP 304 (Not Modified) response with no
+response body. If the Server has newer code available than what is captured by the
+`If-None-Match` header value, or if no `If-None-Match` header is provided in the request,
+the Server parses manifests again. Assuming the resulting payload is different than a
+previous request's, the server provides a different Etag value and new class information
+in the response payload.
 
-Note that the server may, in cases where the client has sent an
-`Accept-Encoding: gzip` HTTP header for the request and the server has chosen
-to provide a gzip-encoded response body, append the characters "--gzip" to
-the end of the Etag.  For example, the HTTP response headers could include:
+If the client sends an `Accept-Encoding: gzip` HTTP header for the request and the Server
+provides a gzip-encoded response body, the Server might append the characters `--gzip` to
+the end of the Etag. For example, the HTTP response headers could include:
 
 ```
-Content-Encoding: gzip
-ETag: e84bbce5482243b3eb3a190e5c90e535cf4f20de--gzip
+Content-Encoding: gzip ETag: e84bbce5482243b3eb3a190e5c90e535cf4f20de--gzip
```

Comment:
I think the `Content-Encoding` and `Etag` header/value pairs should be on separate lines - line break between the words `gzip` and `Etag`.